runtime/j9vm/jvm.c

```diff
@@ -553,6 +558,70 @@ captureCommandLine(void)
 		}
 		close(fd);
 	}
+#elif defined(OSX) /* defined(AIXPPC) */
+	int argmax = 0;
+	size_t length = 0;
+	int mib[3];
+
+	/* query the argument space limit */
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_ARGMAX;
+	length = sizeof(argmax);
+	if (0 == sysctl(mib, 2, &argmax, &length, NULL, 0)) {
+		char *buffer = malloc(argmax);
+
+		if (NULL != buffer) {
+			int argc = 0;
+			int pid = getpid();
+
+			/* query the argument count */
+			mib[0] = CTL_KERN;
+			mib[1] = KERN_PROCARGS2;
+			mib[2] = pid;
+			length = argmax;
+			if (0 == sysctl(mib, 3, buffer, &length, NULL, 0)) {
+				memcpy(&argc, buffer, sizeof(argc));
```

Comment:
Could sizeof(argc) end up copying garbage from the buffer? Why not pass &argc to sysctl() instead of buffer? In theory argmax could be less than 4 causing this to fail for no good reason.

Reply:
Seems weird to depend on the size of argmax to read argc, unless this is actually a requirement. If argmax is too small then we can't get the command line. Is there some small buffer size that can be declared on the stack that does work?

Reply:
`argmax` was about 256k in my experiments, so I don't expect the added condition to ever fail.

Notice that this use of `sysctl` (with `KERN_PROCARGS2`) writes `argc` at the beginning of `buffer`. Unfortunately, it doesn't appear to also provide the argument strings (I got only NULs), which is why there's another call to `sysctl` with `KERN_PROCARGS`.